pkg/controller/statusmanager/status_manager.go

```diff
@@ -261,22 +301,45 @@ func (status *StatusManager) SetFromPods() {
 			continue
 		}
 
+		// Update our last-seen-state if new or changed
+		if dsState == nil {
```

Comment:
Does using a `*daemonState` rather than a `daemonState` buy you anything here? (ie, if you have a zero-initialized `dsState` then is `reflect.DeepEqual(ds.Status, dsState.lastSeenStatus)` guaranteed to be `false` and so you'd hit the next clause anyway?)

Reply:
I thought about that for a while; I decided I hate zero values (especially for time) and didn't want to think about if a zero-value Status is possible.